zulipterminal/ui.py

```diff
@@ -177,15 +177,21 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
             self.controller.editor_mode = True
             self.controller.editor = self.user_search
             return key
-        elif is_command_key('SEARCH_STREAMS', key):
+        elif (is_command_key('SEARCH_STREAMS', key) or
+                is_command_key('SEARCH_TOPICS', key)):
             # jump stream search
             self.body.focus_position = 0
             self.left_panel.keypress(size, 'q')
             self.show_right_panel(visible=False)
             self.show_left_panel(visible=True)
-            self.stream_w.stream_search_box.set_edit_text("")
-            self.controller.editor_mode = True
-            self.controller.editor = self.stream_w.stream_search_box
+            if self.left_panel.is_in_topic_view:
+                self.topic_w.topic_search_box.set_edit_text("")
+                self.controller.editor_mode = True
+                self.controller.editor = self.topic_w.topic_search_box
+            else:
+                self.stream_w.stream_search_box.set_edit_text("")
+                self.controller.editor_mode = True
+                self.controller.editor = self.stream_w.stream_search_box
```

Comment:
The only difference here is which box, so maybe just set a variable in the conditional and do the common actions after that?

Reply:
Sounds fair :+1: